### PR TITLE
Added keyserver-options http-proxy to apt-key command in the install script

### PR DESCRIPTION
**Pull request by moredatapls**
**\- What I did, how I did it & description for the changelog**
I added the usage of `$http_proxy` to the install script as mentioned in #20022.

**-How to verify it**
Occasionally seems to be an issue for some users that want to install Docker with the install script behind a proxy. Tested it on Ubuntu 14.04.4 with and without `$http_proxy` set, and seems to work fine for me so far. Good thing is that if the proxy is not set, apt-key just ignores the option. This means that no manual check in the script is required.

Signed-off-by: Helge Bruegner helge@bruegner.de

### CODE REVIEW

**Comment by fkautz** [comment]
@hbrgnr Your commit needs to be signed.

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

**Comment by thaJeztah** [comment]
ping @hbrgnr can you rebase, squash, and sign off your commit, using the instructions that @GordonTheTurtle showed above, otherwise we can't accept your PR

feel free to ask for help in the #docker-dev IRC channel if you need help doing so

**Comment by hbrgnr** [comment]
@thaJeztah @fkautz sorry, haven't had time to look at it until today. git wanted me to sign 275 commits after I did an "update branch", that's why I just reverted the branch back to my second commit now.

what can I do for "userns" and "windowsTP5" to pass? the error messages seem to be unrelated to my change

File: hack/install.sh [hunk 1]
```diff
@@ -425,7 +425,7 @@ do_install() {
 			(
 			set -x
 			for key_server in $key_servers ; do
-				$sh_c "apt-key adv --keyserver hkp://${key_server}:80 --recv-keys ${gpg_fingerprint}" && break
+				$sh_c "apt-key adv --keyserver hkp://${key_server}:80 --keyserver-options http-proxy=$http_proxy --recv-keys ${gpg_fingerprint}" && break
```
Comment (thaJeztah) [inline hack/install.sh]: Is it a problem that this is added unconditionally? Should we add it *only* if a `$http_proxy` or `$HTTP_PROXY` is set?

Comment (hbrgnr) [inline hack/install.sh]: When I tested it, `apt-key` ignored this option when neither of these variables was set. But yeah, if only `$HTTP_PROXY` (uppercase) is set, this change is not sufficient

**Comment by hbrgnr** [comment]
This PR was meant to resolve #20022, but apparently people seem to experience other issues with the keyserver (see [here](https://github.com/docker/docker/issues/20022#issuecomment-240080273)). Not sure if this is enough then

File: hack/install.sh (hunk 1 above)
Comment (tianon) [inline hack/install.sh]: It doesn't pick these up automatically? :disappointed: :cry:

**Comment by vdemeester** [comment]
@hbrgnr hum seems like #20022 is closed now, is this PR still needed :sweat_smile: ?

**Comment by hbrgnr** [comment]
@vdemeester Nope, close it, glad this is resolved :)

**Comment by vdemeester** [comment]
@hbrgnr cool, thanks for the contribution :stuck_out_tongue_closed_eyes:

**Pull request closed without merging**